server/test/observations.spec.js

```diff
@@ -0,0 +1,176 @@
+        maxParticipants: 10
+      });
+    }).then(function (event) {
+      // Add shepherd to event
+      eventInstance = event;
+      eventId = event.id;
+      return shepherdInstance.addShepherdEvent(event);
+    }).then(function() {
+      // Create sheep
+      return db.User.create({
+        username: 'Lonely Sheep'
+      });
+    }).then(function(sheep) {
+      // Add sheep to event
+      sheepInstance = sheep;
+      sheepId = sheep.id;
+      return sheep.addSheepEvent(eventInstance);
+    }).then(function() {
+      // Create observation
+      db.Observation.create({
+        content: 'These cats are crazy',
+        completed: false,
+        UserId: sheepId,
+        EventId: eventId
+      }).then(function(observation) {
+        // Associate the observation with the event
+        observationInstance = observation;
+        observationId = observation.id;
+        return eventInstance.addObservation(observation);
+      }).then(function() {
+        // Associate the observation with the sheep
+        return sheepInstance.addObservation(observationInstance);
+      }).then(function() {
+        done();
+      });
+    });
+  });
+});
+
+// Delete test observation, event, and users from db after tests
+after(function(done) {
+  observationInstance.destroy().then(function() {
+    eventInstance.destroy().then(function() {
+      sheepInstance.destroy().then(function() {
+        shepherdInstance.destroy().then(function() {
+          done();
+        });
+      });
+    });
+  });
+});
+
+describe('Observation API Endpoints', function() {
+  var basePath = '/api/observation';
+
+  describe('/api/observation/:eventId', function() {
+    it('Should return an array with the observation on GET for a valid eventId', function(done) {
+      request(app)
+        .get(basePath + '/' + eventId)
+        .set('Accept', 'application/json')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect(function(res) {
+          if (!Array.isArray(res.body))
+            throw new Error('Expected an array, received ' + JSON.stringify(res.body));
+        })
+        .expect(function(res) {
+          if (res.body[0].id !== observationId)
+            throw new Error('Expected observation ID ' + observationId + ', received ' + res.body[0].id);
+        })
+        .end(done);
+    });
+
+    it('Should return an empty array on GET for an invalid eventId', function(done) {
+      request(app)
+        .get(basePath + '/0')
+        .expect('Content-Type', /json/)
+        .expect(200)
+        .expect(function(res) {
+          if (res.body.length > 0) {
+            throw new Error('Expected an empty array, received ' + JSON.stringify(res.body));
+          }
+        })
+        .end(done);
+    });
+  });
+
+  describe('/api/observation/create', function() {
+    it('Should create a new observation on POST', function(done) {
+      var observation = {content: 'Smelly cats!!!', userId: sheepId, eventId: eventId};
```

Comment:
might have to refactor this with the addition of rawImage column from the photos. todo later